addons/account/views/report_invoice.xml

```diff
@@ -365,7 +365,8 @@
         </template>
 
         <template id="document_tax_totals_company_currency_template">
-            <t t-set="show_company_taxes" t-value="o.company_id.currency_id != o.currency_id and o.amount_tax != 0 and o.move_type in ['out_invoice', 'out_refund']"/>
+            <t t-set="show_company_taxes" t-value="o.company_id.currency_id != o.currency_id and o.amount_tax != 0 and
+                o.move_type in ['out_invoice', 'out_refund'] and o.company_id.display_invoice_tax_company_currency"/>
```

Comment:
I'm not sure if we could short circuit those other conditions if we place `o.company_id.display_invoice_tax_company_currency` first. 
Also, Might as well put each condition on a new line starting with `and`